### ISSUE TITLE
Re-Auth needed often after reboot of HA

### ISSUE THREAD

**Issue by vistalba** [HomeAssistant-Mods/home-assistant-miele#167]
I really don't know why but since multiple month (maybe 6 or more?) I most of the time have to re-auth the plugin with my login. 
The Token file is there and the token is not expired at the time of the reboot. But after rebooting HA I do get a notification to re-auth.

Anyone else having similar issues? Any hint how to nail this issue down? It is quite annoying. Thx.

**Comment by itchensen** [comment]
Can confirm this issue since several month. Would be great, to have a solution or at least workaround.

**Comment by ljenzen** [comment]
having the same issue - solution welcome

**Comment by larstobi** [comment]
Same

**Comment by mKenfenheuer** [comment]
Same here. Also with the cache path set:

```yaml
miele: 
    cache_path: /homeassistant/miele.json
```

I can also see the cache file being filled/updated.
However occasionally this doesn't seem to work:

```log
2024-02-13 12:38:13.419 WARNING (SyncWorker_7) [custom_components.miele.miele_at_home] Couldn't write token cache to /homeassistant/miele.json
```

**Comment by sHedC** [comment]
Hi,

I am working on moving this to config entries, which is more work than expected.  But nearly done, just fixing services and updating Readme then will submit a pull request.

The new one uses the oauth built in and seems more stable in that respect, just had some issues with the miele.<device> and services which I am fixing.

**Comment by x3style** [comment]
> Hi,
> 
> I am working on moving this to config entries, which is more work than expected. But nearly done, just fixing services and updating Readme then will submit a pull request.
> 
> The new one uses the oauth built in and seems more stable in that respect, just had some issues with the miele. and services which I am fixing.

Hi mate, any chance you are still working on this?

**Comment by sHedC** [comment]
> > Hi,
> > I am working on moving this to config entries, which is more work than expected. But nearly done, just fixing services and updating Readme then will submit a pull request.
> > The new one uses the oauth built in and seems more stable in that respect, just had some issues with the miele. and services which I am fixing.
> 
> Hi mate, any chance you are still working on this?

No, I completed it put a pull request which was ignored and I see no real activity on ths, in the mean time I found this one https://github.com/astrandb/miele which is in HACS and works better so using that and will contribute to that if I see anything that doesn't work for my dryer and washing machine.

**Comment by x3style** [comment]
> > > Hi,
> > > I am working on moving this to config entries, which is more work than expected. But nearly done, just fixing services and updating Readme then will submit a pull request.
> > > The new one uses the oauth built in and seems more stable in that respect, just had some issues with the miele. and services which I am fixing.
> > 
> > 
> > Hi mate, any chance you are still working on this?
> 
> No, I completed it put a pull request which was ignored and I see no real activity on ths, in the mean time I found this one https://github.com/astrandb/miele which is in HACS and works better so using that and will contribute to that if I see anything that doesn't work for my dryer and washing machine.

Thank you so much!
